config.c

```diff
@@ -727,6 +727,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
     char **scriptDest = NULL;
     struct logInfo *newlog = defConfig;
     char *start, *chptr;
+    char *compresscmd_base;
```

Comment:
Please move the variable definition to the innermost code block to limit its scope.